test/forms/TopupPeriodsFormProviderSpec.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms
+
+import java.time.LocalDate
+
+import forms.behaviours.CheckboxFieldBehaviours
+import org.scalacheck.{Gen, Shrink}
+import generators.Generators
+import org.scalatestplus.scalacheck.ScalaCheckPropertyChecks
+import play.api.data.FormError
+
+class TopupPeriodsFormProviderSpec extends CheckboxFieldBehaviours with Generators with ScalaCheckPropertyChecks {
+
+  val form = new TopupPeriodsFormProvider()()
+
+  implicit val noShrink: Shrink[Int] = Shrink.shrinkAny
+
+  ".value" must {
+
+    val fieldName = "value"
+    val requiredKey = "topupPeriods.error.required"
+
+    "bind valid values" in {
+      val dateGen = datesBetween(
+        LocalDate.of(2020, 3, 1),
+        LocalDate.of(2020, 6, 30)
+      )
+
+      val listGen = for {
+        length <- Gen.chooseNum(1, 10)
+        list   <- Gen.listOfN(length, dateGen)
+      } yield list
+
+      forAll(listGen) { list =>
+        val data = list.zipWithIndex.map(item => s"$fieldName[${item._2}]" -> item._1.toString).toMap
+        form.bind(data).get shouldEqual list
```

Comment:
Could use `.value` from `OptionValues` again.

Reply:
That form.bind(data) returns a Form[List[LocalDate]], I haven't amended this one